Request need to send when order complete

vigneshrahane opened the pull request.

thamizhsoftsuave committed:
Request need to send when order complete

thamizhsoftsuave committed:
Get all product list

karthikraja34 commented on admin/admin-menu-loader.php:
```diff
@@ -1,12 +1,65 @@
+
+   function admin_menu_main() {
+
+         get_products();
```
Getter should always return something

karthikraja34 replied:
Format whole code

karthikraja34 commented on admin/admin-menu-loader.php:
```diff
@@ -1,12 +1,65 @@
 
-   ?>
\ No newline at end of file
+   function get_products() {
```
This is not returning products. This is returning view

karthikraja34 commented on admin/admin-menu-loader.php:
```diff
@@ -1,12 +1,65 @@
+
+   }
+
+      function product_table($productslist) {
+
+       ?>  <div class="wrap">
+                 <h2><?php esc_attr_e( 'Products', 'WpAdminStyle' ); ?></h2>
+                 <table class="widefat">
+                 	<thead>
+                 	<tr>
+                 		<th class="row-title"><?php esc_attr_e( 'Icon' ); ?></th>
+                 		<th><?php esc_attr_e( 'Name' ); ?></th>
+                 		<th><?php esc_attr_e( 'SKU' ); ?></th>
+                 		<th><?php esc_attr_e( 'Price' ); ?></th>
+                 		<th><?php esc_attr_e( 'Stock' ); ?></th>
+                 	</tr>
+                 	</thead>
+                 	<tbody>
+                 	<?
+                 	while ( $productslist->have_posts() ) : $productslist->the_post();
+                                     global $product; ?>
+                 	<tr>
+                 	<td class="row-title"><label for="tablecell"><span class='wp-post-image'><?php echo woocommerce_get_product_thumbnail(); ?></span></label></td>
+                     <td class="row-title"><label for="tablecell"><?php echo get_the_title(); ?></label></td>
+                     <td class="row-title"><label for="tablecell"><?php echo  $product->get_sku(); ?></label></td>
+                     <td class="row-title"><label for="tablecell"><?php echo  $product->get_regular_price(); ?></label></td>
+                     <td class="row-title"><label for="tablecell"><?php echo  $product->get_stock_status(); ?></label></td>
+                 	</tr>
+                 	 <? endwhile; ?>
+                 	</tbody>
+                 </table>
+                 </div>
+      <?
+      }
+
+   ?>
```
This violates SRP principle.

karthikraja34 commented on admin/complete-order.php:
```diff
@@ -0,0 +1,45 @@
+<?php
+
+         add_action('woocommerce_thankyou',  'order_completed', 10, 1);
```
This is not the right place.

karthikraja34 commented on admin/complete-order.php:
```diff
@@ -0,0 +1,45 @@
+
+         add_action('woocommerce_thankyou',  'order_completed', 10, 1);
+
+        function order_completed($order_id)
```
Function names should always be verb

karthikraja34 commented on testpress-lms.php:
```diff
@@ -64,6 +64,7 @@ function deactivate_testpress_lms() {
  */
 require plugin_dir_path( __FILE__ ) . 'includes/class-testpress-lms.php';
 require plugin_dir_path( __FILE__ ) . 'admin/admin-menu-loader.php';
+require plugin_dir_path( __FILE__ ) . 'admin/complete-order.php';
```
Change appropriate filename

thamizhsoftsuave committed:
Update pr review changes

--Formatted code
--modified old function names with appropriate names
--added comments for functionality
